internal/linereader/multi_reader.go

```diff
@@ -29,6 +29,33 @@ const (
 	InputFormatLP
 )
 
+func (i *InputFormat) Set(v string) error {
+	switch v {
+	case "":
+		*i = InputFormatDerived
+	case "lp":
+		*i = InputFormatLP
+	case "csv":
+		*i = InputFormatCSV
+	default:
+		return fmt.Errorf("unsupported format: %q", v)
+	}
+	return nil
+}
```

Comment:
We implement `cli.Generic` and then use the `cli.GenericFlag` flag to bind to these strongly-typed flags. These flags parse and validate the values before calling our individual command action functions.